packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -457,46 +492,53 @@ where
             ..Default::default()
         };
 
+        let latest_signature_slot = headers.last().map(|h| h.consensus_update.signature_slot);
+
         // Final check to make sure the target chain's calculated slot is greater than our latest
-        // update slot:
+        // update's signature slot
         wait_for_condition(
             Duration::from_secs(15 * 60),
             Duration::from_secs(10),
             || async {
+                if headers.is_empty() {
+                    return Ok(true);
+                }
+
                 let latests_tm_block = self.tm_client.latest_block().await?;
                 let latest_onchain_timestamp = latests_tm_block.block.header.time.unix_timestamp();
                 let calculated_slot = ethereum_client_state
                     .compute_slot_at_timestamp(latest_onchain_timestamp.try_into().unwrap())
                     .unwrap();
```

Comment:
There are two usages `unwrap` here. You should probably walk over all usages of `unwrap` and remove them since it can shutdown the relayer api

Reply:
These should not really be possible to happen, unless something is very wrong.
Besides, a panic here would not shut down the relayer, only the current request